File: leetcode/1535. Find the Winner of an Array Game/main.go
package main

func getWinner(arr []int, k int) int {
	if k == 1 {
		return max(arr[0], arr[1])
	}
	if k >= len(arr) {
		return maxInSlice(arr)
	}

	winner := arr[0]
	consecutiveWins := 0

	for i := 1; i < len(arr); i++ {
		if winner > arr[i] {
			consecutiveWins++
		} else {
			winner = arr[i]
			consecutiveWins = 1
		}

		if consecutiveWins == k {
			return winner
		}
	}

	return winner
}

func maxInSlice(arr []int) int {
	num := 0

	for _, v := range arr {
		num = max(num, v)
	}

	return num
}

File: leetcode/1535. Find the Winner of an Array Game/main.ts
const getWinner = (arr: number[], k: number): number => {
  if (k >= arr.length) {
    return Math.max(...arr)
  }

  let winner = arr[0]
  let consecutiveWins = 0

  for (let i = 1; i < arr.length; i++) {
    if (winner > arr[i]) {
      consecutiveWins++
    } else {
      winner = arr[i]
      consecutiveWins = 1
    }

    if (consecutiveWins == k) {
      break
    }
  }

  return winner
}
